File: sqry-lang-rust/tests/fixtures/rust/simple_calls.rs
// Test fixture: Basic function calls
// Tests: main→greet, process_data→fetch→transform

fn greet(name: &str) -> String {
    format!("Hello, {}!", name)
}

fn fetch(id: u32) -> Option<String> {
    if id > 0 {
        Some(format!("Data-{}", id))
    } else {
        None
    }
}

fn transform(data: String) -> String {
    data.to_uppercase()
}

fn process_data(id: u32) -> Option<String> {
    let raw = fetch(id)?;
    let processed = transform(raw);
    Some(processed)
}

fn main() {
    let message = greet("World");
    println!("{}", message);

    if let Some(result) = process_data(42) {
        println!("Processed: {}", result);
    }
}

File: sqry-lang-java/tests/fixtures/java/com/example/simple/SimpleCalls.java
// Test fixture: Basic method calls
// Tests: main→greet, processData→fetch→transform

package com.example.simple;

public class SimpleCalls {

    public static String greet(String name) {
        return String.format("Hello, %s!", name);
    }

    public static String fetch(int id) {
        if (id > 0) {
            return String.format("Data-%d", id);
        }
        return "";
    }

    public static String transform(String data) {
        return data.toUpperCase();
    }

    public static String processData(int id) {
        String raw = fetch(id);
        if (!raw.isEmpty()) {
            return transform(raw);
        }
        return "";
    }

    public static void main(String[] args) {
        String message = greet("World");
        System.out.println(message);

        String result = processData(42);
        if (!result.isEmpty()) {
            System.out.println("Processed: " + result);
        }
    }
}
